spec/index.bs

```diff
@@ -2612,6 +2613,140 @@ If a format is listed in [[#plain-color-formats]] with {{GPUTextureUsage/RENDER_
 color renderable format. Any other format is not a color renderable format. Any depth/stencil format is a
 depth or stencil renderable format. Any other format is not a depth or stencil renderable format.
 
+## <dfn interface>GPUExternalTexture</dfn> ## {#gpu-external-texture}
+
+A {{GPUExternalTexture}} is a sampleable texture wrapping an external video object.
+The contents of a {{GPUExternalTexture}} object may not change, either from inside WebGPU
+(it is only sampleable) or from outside WebGPU (e.g. due to video frame advancement).
+
+Issue: Update this description with canvas.
+
+They are bound into bind group layouts using the {{GPUBindGroupLayoutEntry/externalTexture}}
+bind group layout entry member.
+Using external textures costs several binding points: see [=binding cost=].
+
+<div class=note>
+    External textures *can* be implemented without creating a copy of the imported source,
+    but this depends implementation-defined factors.
+    Ownership of the underlying representation may either be exclusive or shared with other
+    owners (such as a video decoder), but this is not visible to the application.
+
+    The underlying representation of an external texture is unobservable
+    (except for sampling behavior) but typically may include
+
+    - Up to three 2D planes of data (e.g. RGBA, Y+UV, Y+U+V).
+    - Metadata for converting coordinates before reading from those planes (crop and rotation).
+    - Metadata for converting values into the specified output color space (matrices, gammas, 3D LUT).
+
+    The configuration used may not be stable across time, systems, user agents, media sources,
+    or frames within a single video source.
+    In order to account for many possible representations,
+    the [=binding cost=] conservatively requires:
+
+    - for pipeline layouts that have *any* external textures, one sampler binding
```

Comment:
This might be difficult to use in Vulkan for example because it's not clear where the extra sampler would go. You can't declare the sampler at BindGroupLayout creation because there might be different bindgroups used by the pipeline each with an external texture.

Maybe we can require two samplers per external_texture binding? It doesn't seem too painful given we already require 4 (!!) texture bindings.

Reply:
Just one sampler per binding, the actual image sampler is user-provided (unless we change this to allow post-conversion filtering!)
So 4 textures + 1 sampler + 1 uniform buffer. Seems OK.